### PR TITLE
feat:  add ifthen expression support

### PR DESCRIPTION
**Pull request by EpsilonPrime**
In this implementation IFTHEN is being treated as a built-in function.

### CODE REVIEW

File: src/substrait/textplan/converter/PlanPrinterVisitor.cpp [hunk 1]
```diff
@@ -442,10 +442,28 @@ std::any PlanPrinterVisitor::visitWindowFunction(
...
+  std::stringstream text;
+  text << "IFTHEN(";
+  bool hasPreviousText = false;
+  for (const auto& clause : ifthen.ifs()) {
+    if (hasPreviousText) {
+      text << ", ";
+    }
+    text << ANY_CAST(std::string, visitExpression(clause.if_()));
+    text << ", ";
+    text << ANY_CAST(std::string, visitExpression(clause.then()));
```
Comment (westonpace) [inline src/substrait/textplan/converter/PlanPrinterVisitor.cpp:454]: What happens if `then` is not set?  Do we crash?  It's ok if it's invalid.

File: src/substrait/textplan/converter/PlanPrinterVisitor.cpp [hunk 2]
```diff
@@ -442,10 +442,28 @@ std::any PlanPrinterVisitor::visitWindowFunction(
...
+    hasPreviousText = true;
+  }
+  if (ifthen.has_else_()) {
+    if (hasPreviousText) {
+      text << ", ";
+    }
+    text << ANY_CAST(std::string, visitExpression(ifthen.else_()));
+    hasPreviousText = true;
```
Comment (westonpace) [inline src/substrait/textplan/converter/PlanPrinterVisitor.cpp:462]: Super minor nit, could remove.
```suggestion
```

**Review by westonpace — CHANGES_REQUESTED** [review]
As discussed offline, maybe add one small test case outside the round trip tests.

File: src/substrait/textplan/converter/PlanPrinterVisitor.cpp (hunk 1 above)
Comment (EpsilonPrime) [reply src/substrait/textplan/converter/PlanPrinterVisitor.cpp:460]: Added check and test for the same.

**Review by westonpace — APPROVED** [review]

**Pull request merged**